### PR TITLE
Triage bot: add unit test step and document shared test utilities

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Adds a new "Write a Unit Test" step to the triage bot's fix skill, so it produces a regression test alongside every fix attempt.
- Creates `reference/unit-testing.md` documenting unit test conventions, file placement, and the shared test utilities/mocks available in the repo. Referenced from `AGENTS.md` so any agent (not just triage) can use it.
- Adds a **Unit Test** line to the triage bot's GitHub comment template.

## Testing

- No test changes — this PR only modifies agent instructions and reference docs.

## Docs

- No user-facing docs needed. `reference/unit-testing.md` is agent-facing only.

### CODE REVIEW

File: .agents/skills/triage/fix.md [hunk 1]
```diff
@@ -109,11 +110,25 @@ Watch for build errors — fix any TypeScript issues before proceeding.
 
 Re-run the reproduction, often using `pnpm run build`/`astro build` or `pnpm run dev`/`astro dev`.
 
-## Step 6: Check for Regressions
+## Step 6: Write a Unit Test
+
+Write a unit test that covers the bug you just fixed. The test should fail without the fix and pass with it. This prevents regressions and documents the expected behavior for maintainers.
+
+Read [`reference/unit-testing.md`](../../reference/unit-testing.md) for conventions, file placement, shared test utilities, and mocks. Name your `it` block after the specific behavior being verified, referencing the issue number for traceability (e.g. `it('handles edge case from issue #1234')`).
```
Comment (Princesseuh) [inline .agents/skills/triage/fix.md]: While the traceability is great, `handles edge case from issue #1234` isn't an amazing name, imo. I think either asking the AI to put a comment with the issue number above a normally named test, or giving an example closer to "handles {problem} (#1234)" would be better?

**Review by Princesseuh — APPROVED** [review]

File: .agents/skills/triage/fix.md (hunk 1 above)
Comment (matthewp) [reply .agents/skills/triage/fix.md]: Oh yeah this is terrible, it should be a descriptive name, will update. I'm not sure it even helps to have the issue number, those can be useful but lose their meaning with time.
